torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -5075,14 +5075,12 @@ c10::DeviceIndex ProcessGroupNCCL::guessDeviceId() const {
   // offset wrt the device id if intra-node GPUs are sharded into multiple
   // dimensions.
   int devIdx = globalRank() % localDeviceCount_;
-  LOG(WARNING)
-      << logPrefix()
-      << c10::str(
-             " using GPU ",
-             devIdx,
-             " as device used by this process is currently unknown. ",
-             "This can potentially cause a hang if this rank to GPU mapping is incorrect. ",
-             "You can specify device_id in init_process_group() to force use of a particular device.");
+  if (rank_ == 0) {
```

Comment:
Should we log once per host? i.e. check `LOCAL_RANK == 0` instead?

In someways if we don't log on every host it makes it very easy for this warning to be ignored

Reply:
Good point, thanks